tests/functional/z_cluster/nodes/test_non_ocs_taint_and_toleration.py

```diff
@@ -227,3 +369,140 @@ def test_non_ocs_taint_and_tolerations(self):
+                f'"mds": {tolerations}, "metrics-exporter": {tolerations}, "noobaa-core": {tolerations}, '
+                f'"rgw": {tolerations}, "toolbox": {tolerations}'
+            )
+            param = f'{{"spec": {{"placement": {{{param}}}}}}}'
+
+        storagecluster_obj.patch(params=param, format_type="merge")
+        logger.info(f"Successfully added toleration to {storagecluster_obj.kind}")
+
+        logger.info("Add tolerations to the subscription")
+        sub_list = ocp.get_all_resource_names_of_a_kind(kind=constants.SUBSCRIPTION)
+        param = (
+            '{"spec": {"config":  {"tolerations": '
+            '[{"effect": "NoSchedule", "key": "xyz", "operator": "Equal", '
+            '"value": "true"}]}}}'
+        )
+        for sub in sub_list:
+            if sub == constants.ODF_SUBSCRIPTION:
+                sub_obj = ocp.OCP(
+                    resource_name=sub,
+                    namespace=config.ENV_DATA["cluster_namespace"],
+                    kind=constants.SUBSCRIPTION,
+                )
+                sub_obj.patch(params=param, format_type="merge")
+                logger.info(f"Successfully added toleration to {sub}")
+        logger.info("Check custom toleration on all subscriptions")
+        retry(CommandFailed, tries=5, delay=10,)(
+            check_toleration_on_subscriptions
+        )(toleration_key="xyz")
+
+        if config.ENV_DATA["mcg_only_deployment"]:
+            logger.info("Wait some time after adding toleration for pods respin")
+            waiting_time = 60
+            logger.info(f"Waiting {waiting_time} seconds...")
+            time.sleep(waiting_time)
+            logger.info("Force delete all pods")
+            pod_list = get_all_pods(
+                namespace=config.ENV_DATA["cluster_namespace"],
+                exclude_selector=True,
+            )
+            for pod in pod_list:
+                pod.delete(wait=False)
+
+        logger.info("After edit noticed few pod respins as expected")
+        assert wait_for_pods_to_be_running(timeout=1200, sleep=15)
+
+        logger.info(
+            "Check custom toleration on all newly created pods under openshift-storage"
+        )
+        retry(AssertionError, tries=5, delay=10)(check_toleration_on_pods)(
+            toleration_key="xyz"
+        )
+
+        if config.DEPLOYMENT["external_mode"]:
+            cephcluster = CephClusterExternal()
+            cephcluster.cluster_health_check()
+        else:
+            self.sanity_helpers.health_check()
+
+        # Get the node list
+        node = get_nodes("worker", num_of_nodes=1)
+
+        # Reboot one master nodes
+        nodes.restart_nodes(node, wait=False)
+
+        # Wait some time after rebooting master
+        waiting_time = 40
+        logger.info(f"Waiting {waiting_time} seconds...")
+        time.sleep(waiting_time)
+
+        # Validate all nodes and services are in READY state and up
```

Comment:
Test step says : 5. Reboot one of the nodes and check toleration on all odf pods on that node.

where are we checking toleration on odf pods after reboots ?